Doc/library/annotationlib.rst

```diff
@@ -0,0 +1,231 @@
+      and constant values optimizations.
+
+      The exact values of these strings may change in future versions of Python.
+
+   .. versionadded:: 3.14
+
+.. class:: ForwardRef
+
+   A proxy object for forward references in annotations.
+
+   Instances of this class are returned when the :attr:`~Format.FORWARDREF` format is
+   used and annotations contain a name that cannot be resolved.  This can happen
+   when a forward reference is used in an annotation, such as when a class is
+   referenced before it is defined.
+
+   .. attribute:: __forward_arg__
+
+      A string containing the code that was evaluated to produce the :class:`~ForwardRef`.
+      The string may not be exactly equivalent to the original source.
+
+   .. method:: evaluate(*, globals=None, locals=None, type_params=None, owner=None)
+
+      Evaluate the forward reference, returning its value.
+
+      This may throw an exception such as :exc:`NameError` if the forward reference
+      refers to names that do not exist. The arguments to this method can be used to
+      provide bindings for names that would otherwise be undefined.
+
+      :class:`~ForwardRef` instances returned by :func:`get_annotations` retain
+      references to information about the scope they originated from, so calling
+      this method with no further arguments may be sufficient to evaluate such objects.
+      :class:`~ForwardRef` instances created by other means may not have any information
+      about their scope, so passing arguments to this method may be necessary to
+      evaluate them successfully.
+
+      *globals* and *locals* are passed to :func:`eval`, representing the global and
+      local namespaces in which the name is evaluated. *type_params*, if given, must be
+      a tuple of :ref:`type parameters <type-params>` that are in scope while the forward
+      reference is being evaluated. *owner* is the object that owns the annotation from
+      which the forward reference derives, usually a function, class, or module.
+
+      .. important::
+
+         Once a :class:`~ForwardRef` instance has been evaluated, it caches the evaluated
+         value, and future calls to :meth:`evaluate` will return the cached value, regardless
+         of the parameters passed in.
+
+   .. versionadded:: 3.14
+
+.. function:: get_annotate_function(obj)
+
+   Retrieve the :term:`annotate function` for *obj*. Return :const:`!None` if *obj* does not have an
+   annotate function.
+
+   This is usually equivalent to accessing the :attr:`~object.__annotate__` attribute of *obj*,
+   but direct access to the attribute may return the wrong object in certain situations involving
+   metaclasses. This function should be used instead of accessing the attribute directly.
+
+   .. versionadded:: 3.14
+
+.. function:: get_annotations(obj, *, globals=None, locals=None, eval_str=False, format=Format.VALUE)
+
+   Compute the annotations dict for an object.
```

Comment:
For my understanding, is there some best practice for when we should use  :class:`dict` vs. just dict?

Reply:
I don't think there's a hard-and-fast rule. Part of me wants to always add a link for consistency, but when I suggested that for :const:`None` in the past, people rightly pushed back that these links probably aren't useful for most readers. One way to think of it: If you're an advanced enough Python programmer to be reading these docs, you probably don't need to be told what a dict is.

In this particular case, the real reason is that I copied the docs from https://docs.python.org/3/library/inspect.html#inspect.get_annotations, which use unlinked "dict".

Reply:
Yeah, that's fair. I was reading and realized that the links for types are used pretty inconsistently across the docs. I couldn't discern if this was backed by some general rules that I was unaware of...but yeah, totally agree. I don't know if it'd be super useful here.

Reply:
@savannahostrowski Yeah there is definitely inconsistency and a bit of personal preference about the distraction of too many links.